src-cpp/app/modules/main/wallet/accounts/controller.cpp

```diff
@@ -22,4 +22,45 @@ QList<Wallets::WalletAccountDto> Controller::getWalletAccounts()
 
     return wallet_accounts;
 }
+
+QString Controller::generateNewAccount(QString password, QString accountName, QString color)
+{
+    if(m_walletServicePtr != nullptr)
+    {
+        return m_walletServicePtr->generateNewAccount(password, accountName, color);
+    }
+}
+
+QString Controller::addAccountsFromPrivateKey(QString privateKey, QString password, QString accountName, QString color)
+{
+    if(m_walletServicePtr != nullptr)
+    {
+        return m_walletServicePtr->addAccountsFromPrivateKey(privateKey, password, accountName, color);
+    }
+}
+
+QString Controller::addAccountsFromSeed(QString seedPhrase, QString password, QString accountName, QString color)
+{
+    if(m_walletServicePtr != nullptr)
+    {
+        return m_walletServicePtr->addAccountsFromSeed(seedPhrase, password, accountName, color);
+    }
+}
+
+QString Controller::addWatchOnlyAccount(QString address, QString accountName , QString color)
+{
+    if(m_walletServicePtr != nullptr)
+    {
+        return m_walletServicePtr->addWatchOnlyAccount(address, accountName, color);
+    }
+}
```

Comment:
what happens if `m_walletServicePtr` is `nullptr`? which value returned?

Reply:
will return an error string, thanks!